cmd/api/src/api/v2/file_uploads.go

```diff
@@ -126,8 +127,12 @@ func (s Resources) ProcessFileUpload(response http.ResponseWriter, request *http
 		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, api.ErrorResponseDetailsIDMalformed, request), response)
 	} else if fileUploadJob, err := fileupload.GetFileUploadJobByID(s.DB, int64(fileUploadJobID)); err != nil {
 		api.HandleDatabaseError(request, response, err)
-	} else if fileName, err := fileupload.SaveToTempFile(s.Config.TempDirectory(), request.Body); err != nil {
-		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
+	} else if fileName, err := fileupload.SaveIngestFile(s.Config.TempDirectory(), request.Body); err != nil {
+		if errors.Is(err, fileupload.InvalidIngestFileType) {
+			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
```

Comment:
We might want to return a status code 415 here but I'm not totally sure because we're checking the mime type of the persisted file and not the payload itself. Unless someone else has some feels I say it's your call.

Reply:
Since this part of the request is concerned with the contents of the request rather than the transport, I opted to handle this how we handle other places where a marshal failure occurs, which is consider it a bad request. We had a team discussion which resulted in creating an internal ticket to track improving the validation step beyond just checking for plain text content.